WordPress/wordpress-develop
# Ensure root selector (body) is not wrapped in :root :where()

Trac ticket: https://core.trac.wordpress.org/ticket/61704
Gutenberg issue: https://github.com/WordPress/gutenberg/issues/63712
Gutenberg PR: https://github.com/WordPress/gutenberg/pull/63726

## What?
In WordPress 6.6, there was a general effort to lower and level specificity, but a couple of selectors seem to have mistakenly had their specificity raised.

The styles associated with the `body` are an example. In 6.5 these styles used the simple `body` selector. In 6.6 the selector is `:root :where(body)`, an increase in specificity.

## Why?
These changes may cause some problems for themes, especially the typography styles in the body, which will cascade.

## How?
Reverts the selector back to `body`

Note that unlike the gutenberg PR, the core backport only affects the frontend styles. The editor styles will be fixed by updating the packages from gutenberg.

*props to @aaronrobertshaw for pointing me to the fix*

## Testing Instructions
1. Open the site editor
2. Click the editor area to open the editor
3. Click 'View site' to view the frontend
4. Insect the body element's styles
5. Using the browser dev tools, check that the styles are like 'expected' snippet below, with a simple `body` selector and not a `:root :where(body)` selector.

Expected:
```css
body {
    --wp--style--root--padding-top: 115px;
    --wp--style--root--padding-right: var(--wp--preset--spacing--50);
    --wp--style--root--padding-bottom: 115px;
    --wp--style--root--padding-left: var(--wp--preset--spacing--50);
    color: var(--wp--preset--color--contrast);
    background-color: var(--wp--preset--color--base);
    font-family: var(--wp--preset--font-family--body);
    font-size: var(--wp--preset--font-size--medium);
    font-style: normal;
    font-weight: 400;
    line-height: 1.55;
}
```

Before:
```css
:root :where(body) {
    --wp--style--root--padding-top: 115px;
    --wp--style--root--padding-right: var(--wp--preset--spacing--50);
    --wp--style--root--padding-bottom: 115px;
    --wp--style--root--padding-left: var(--wp--preset--spacing--50);
    color: var(--wp--preset--color--contrast);
    background-color: var(--wp--preset--color--base);
    font-family: var(--wp--preset--font-family--body);
    font-size: var(--wp--preset--font-size--medium);
    font-style: normal;
    font-weight: 400;
    line-height: 1.55;
}
```

---
**This Pull Request is for code review only. Please keep all other discussion in the Trac ticket. Do not merge this Pull Request. See [GitHub Pull Requests for Code Review](https://make.wordpress.org/core/handbook/contribute/git/github-pull-requests-for-code-review/) in the Core Handbook for more details.**
